app/Helpers/database/subscription.php

```diff
@@ -25,74 +26,16 @@ function updateSubscription(string $subjectType, int $subjectId, int $userId, bo
 }
 
 /**
- * @deprecated $implicitSubscriptionQry considered harmful. Use Eloquent ORM.
- *
- * Checks whether a given user is subscribed to a subject, whether implicitly or explicitly.
- *
- * @param string|null $implicitSubscriptionQry optional sql query capable of identifying the existence of an implicit
- *                                         subscription to the subject (must be usable inside an EXISTS clause)
+ * Checks whether a given user is subscribed to a subject explicitly.
  */
-function isUserSubscribedTo(string $subjectType, int $subjectID, int $userID, ?string $implicitSubscriptionQry = null): bool
+function isUserSubscribedTo(string $subjectType, int $topicID, int $userID): bool
 {
-    if (!$userID) {
-        return false;
-    }
-    sanitize_sql_inputs($subjectType);
-
-    if ($implicitSubscriptionQry === null) {
-        $query = "
-            SELECT 1
-            FROM subscriptions
-            WHERE
-              subject_type = '$subjectType'
-              AND subject_id = $subjectID
-              AND user_id = $userID
-              AND state = 1
-        ";
-    } else {
-        // either there's an explicit subscription...
-        // ...or there's an implicit subscription without an explicit unsubscription
-        // optional sql query capable of identifying the existence of an implicit
-        // subscription to the subject (must be usable inside an EXISTS clause)
-        $query = "
-            SELECT 1
-            FROM subscriptions
-            WHERE
-              EXISTS (
-                SELECT 1
-                FROM subscriptions
-                WHERE
-                  subject_type = '$subjectType'
-                  AND subject_id = $subjectID
-                  AND user_id = $userID
-                  AND state = 1
-              )
-              OR (
-                  EXISTS (
-                    $implicitSubscriptionQry
-                  )
-                  AND NOT EXISTS (
-                    SELECT 1
-                    FROM subscriptions
-                    WHERE
-                      subject_type = '$subjectType'
-                      AND subject_id = $subjectID
-                      AND user_id = $userID
-                      AND state = 0
-                  )
-              )
-        ";
-    }
+    $explicitSubcription = Subscription::where('subject_type', $subjectType)
+        ->where('subject_id', $topicID)
+        ->where('user_id', $userID)
+        ->first();
 
-    $dbResult = s_mysql_query($query);
-    if (!$dbResult) {
-        return false;
-    }
-
-    $isSubscribed = mysqli_num_rows($dbResult) > 0;
-    mysqli_free_result($dbResult);
-
-    return $isSubscribed;
+    return $explicitSubcription && ($explicitSubcription->state == 1);
```

Comment:
`->exists()` tends to be more performant than `->first()`, as it only checks existence and doesn't fetch a record/return a record.

```php
return Subscription::where('subject_type', $subjectType)
    ->where('subject_id', $topicID)
    ->where('state', true)
    ->exists();
```

Reply:
For this particular query it seems to be about 0.02ms on my machine, but it makes sense, so I've made the change anyway.